#!/bin/sh -e
## 50_apache-mode.dpatch by Peter S Galbraith <psg@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Support defcustom, add auto-loads, fix font-lock.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/psg/emacs/emacs-goodies-el/newfiles/emacs-goodies-el/elisp/emacs-goodies-el/apache-mode.el emacs-goodies-el/elisp/emacs-goodies-el/apache-mode.el
--- /home/psg/emacs/emacs-goodies-el/newfiles/emacs-goodies-el/elisp/emacs-goodies-el/apache-mode.el	2003-11-06 22:28:23.000000000 -0500
+++ emacs-goodies-el/elisp/emacs-goodies-el/apache-mode.el	2003-11-09 20:49:44.000000000 -0500
@@ -1,10 +1,10 @@
-;;; apache-mode.el --- major mode for editing Apache configuration files
+;;; Apache-mode.el --- major mode for editing Apache configuration files
 
 ;; Author:    Jonathan Marten  <jonathan.marten@uk.sun.com> or
 ;; Author:    Kevin A. Burton  <burton@openprivacy.org>
 ;; Keywords:    languages, faces
-;; Last edit: 
-;; Version: 1.0.1 
+;; Last edit:
+;; Version: 1.1
 
 ;; This file is an add-on for XEmacs or GNU Emacs (not tested with the latter).
 ;;
@@ -44,11 +44,16 @@
 
 ;;; History:
 
+;; V1.1 Peter S Galbraith <psg@debian.org>
+;;  - Don't force font-lock-mode in apache-mode.
+;;  - create customization defgroup and change `apache-mode-hook' to defcustom.
+;;  - add auto-mode-alist entries as autoloads.
+
 ;; Fri Mar  2 17:53:40 2001 (burton@relativity.yi.org): added NameVirtualHost
-;; keyword.  
+;; keyword.
 
 ;; Thu Mar  1 20:16:02 2001 (burton@relativity.yi.org): added Order and Allow
-;; keywords. 
+;; keywords.
 
 ;; Thu Mar  1 20:07:37 2001 (burton@relativity.yi.org): tested under GNU Emacs
 ;; 20.x and fixed font-lock support.
@@ -62,16 +67,21 @@
 ;; Requires
 (require 'regexp-opt)
 
+(defgroup apache-mode nil
+  "Major mode for editing Apache configuration files."
+  :group 'programming)
+
+(defcustom apache-mode-hook nil
+  "*List of hook functions run by `apache-mode' (see `run-hooks')."
+  :type 'hook
+  :group 'apache-mode)
 
 ;; Variables
 (defvar apache-mode-map nil
-  "Keymap used in Apache config mode buffers")
+  "Keymap used in Apache config mode buffers.")
 
 (defvar apache-mode-syntax-table nil
-  "Apache config mode syntax table")
-
-(defvar apache-mode-hook nil
-  "*List of hook functions run by `apache-mode' (see `run-hooks')")
+  "Apache config mode syntax table.")
 
 
 ;; Font lock
@@ -195,16 +205,18 @@
   (make-local-variable 'comment-column)
   (setq comment-column 48)
 
-  ;;font-lock stuff (may not be initially necessary for XEmacs)
-  (font-lock-mode 1)
-  (setq font-lock-keywords apache-font-lock-keywords)
-  (font-lock-fontify-buffer)
+  (set (make-local-variable 'font-lock-defaults)
+       '((apache-font-lock-keywords) t t))
   
   (setq mode-name "Apache")
   (setq major-mode 'apache-mode)
 
   (run-hooks 'apache-mode-hook))
 
+;;;###autoload(add-to-list 'auto-mode-alist '("\\.htaccess$"   . apache-mode))
+;;;###autoload(add-to-list 'auto-mode-alist '("httpd\\.conf$"  . apache-mode))
+;;;###autoload(add-to-list 'auto-mode-alist '("srm\\.conf$"    . apache-mode))
+;;;###autoload(add-to-list 'auto-mode-alist '("access\\.conf$" . apache-mode))
 
 ;; Provides
 (provide 'apache-mode)
